sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -136,8 +139,22 @@ class AstBuilder extends DataTypeAstBuilder
       label: Option[String],
       allowVarDeclare: Boolean): CompoundBody = {
     val buff = ListBuffer[CompoundPlanStatement]()
+    val handlers = ListBuffer[ErrorHandler]()
+    val conditions = mutable.HashMap[String, String]()
+    val sqlStates = mutable.Set[String]()
+
     ctx.compoundStatements.forEach(compoundStatement => {
-      buff += visit(compoundStatement).asInstanceOf[CompoundPlanStatement]
+      val stmt = visit(compoundStatement).asInstanceOf[CompoundPlanStatement]
+
+      stmt match {
+        case handler: ErrorHandler => handlers += handler
+        case condition: ErrorCondition =>
+          assert(!conditions.contains(condition.conditionName))   // Check for duplicate names.
+          assert(!sqlStates.contains(condition.value))            // Check for duplicate sqlStates.
```

Comment:
should these be asserts or exception throws? this seems like an user error?

Reply:
We can throw exceptions here. I will change it now.